src/components/List/List.js

```diff
@@ -0,0 +1,162 @@
+import { db, app, getDatabase, ref, set, get, child } from "../../services/Firebase";
+import { useNavigate } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import Header from './Header/Header';
+import Options from './Options/Options';
+import Body from './Body/Body';
+import Footer from './Footer/Footer';
+import classes from './List.module.css';
+
+
+const List = () => {
+
+    //STATES & HOOKS------------------------------------------------------------------
+    const [originalDataArray, setOriginalDataArray] = useState([]); //this array contains filtered group of data from the originalArray
+    const [filteredDataArray, setFilteredDataArray] = useState([]); //this array contains filtered group of data from the originalArray
+    const [totalCards, setTotalCards] = useState(0); //contains the total number of events' cards
+    const [cardsColorsArray, setCardsColorsArray] = useState([]);
+    const [change, setChange] = useState(false);
+    const navigate = useNavigate();
+
+
+    useEffect(() => {
+        loadEvents(); //Initially load all the events from the database
+    }, []);
+
+    useEffect(() => {
+        if (change) {
+            displayCards(); //load the events from the database to the originalDataArray
+            setChange(false);
+        }
+    });
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    const newButtonClickHandler = () => {
+        set(ref(db, 'eventToEdit/'), { //when user clicks new - no events to edit so make sure eventToEdit is set to "" empty string 
+            id: "",
+            name: "",
+            description: "",
+            date: ""
+        });
+        navigate('/form', { state: { id: 1, headerText: 'Create Event', buttonText: 'Add' } }); //navigate To Form Page
```

Comment:
instead of sending those information to the route state,
please create 2 routes and add prop isNew (default value is true) to the Form component
/form -> component <Form/>
/form:id -> component <Form isNew={false}/>

then in Form component you can minuplate the headerText, buttonText, ... based of mode (isNew)

Reply:
done